File: a005.py
# https://replit.com/@andy8787main/a005#main.py
# https://zerojudge.tw/ShowProblem?problemid=a005

times = eval(input())
for t in range(times):
    l = [int(i) for i in input().split(' ')] # 轉數字
    if (l[1]-l[0]) == (l[2]-l[1]): # 等差
        d = l[1]-l[0] # 等差
        print(" ".join([str(i) for i in l]),end=" ") # join裡的list是要用str
        print(l[3]+d)
    else:
        d = int(l[1]/l[0]) # 等比
        print(" ".join([str(i) for i in l]),end=" ") # join裡的list是要用str
        print(l[3]*d)

File: a005.cpp
// https://replit.com/@andy8787main/a005-cpp#main.cpp
// https://zerojudge.tw/ShowProblem?problemid=a005
#include <iostream>
using namespace std;
int main() {
    int times;
    cin >> times;
    for (int t = 0; t < times; t++) {
        int arr[5];
        for (int i = 0; i < 4; i++) {
            cin >> arr[i];
        }

        if ((arr[2]-arr[1]) == (arr[1]-arr[0])) {
            arr[4] = arr[3]+(arr[2]-arr[1]);
        } else {
            arr[4] = arr[3]*(arr[2]/arr[1]);
        }

        for (auto i:arr){
            cout << i << " ";
        }
        cout << endl;
    }
} 
